components/forms/PostForm.js

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect, useState } from 'react';
+import { useRouter } from 'next/router';
+import Form from 'react-bootstrap/Form';
+import FloatingLabel from 'react-bootstrap/FloatingLabel';
+import PropTypes from 'prop-types';
+import { Button } from 'react-bootstrap';
+import { getAllCategories } from '../../api/tagData';
+import { createPost, updatePost } from '../../api/postsData';
+
+const initialState = {
+  id: 0,
+  userId: 0,
+  categoryId: 0,
+  title: '',
+  publicationDate: '',
+  imageUrl: '',
+  content: '',
+  approved: false,
+};
+
+function PostForm({ obj }) {
+  const [formInput, setFormInput] = useState(initialState);
+  // const [tags, setTags] = useState([]);
+  const [categories, setCategories] = useState([]);
+  const router = useRouter();
+
```

Comment:
Probably create a state for users to store all users like what was done for categories.